ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_core.cc

```diff
@@ -379,16 +440,42 @@ TFCore::run (const GstTensorMemory * input, GstTensorMemory * output)
 
   std::vector <std::pair <string, Tensor>> input_feeds;
   std::vector <Tensor> outputs;
+  Tensor in;
 
   for (int i = 0; i < inputTensorMeta.num_tensors; ++i) {
-    Tensor in (input_tensor_info[i].type, input_tensor_info[i].shape);
-
-    /* copy data */
-    if (input_tensor_info[i].type == DT_STRING) {
-      in.scalar<string>()() = string ((char *) input[i].data, input[i].size);
-    } else {
-      std::copy_n ((char *) input[i].data, input[i].size,
-          const_cast<char *>(in.tensor_data().data()));
+
+    if (mem_optmz) {
+      TFBuffer *buf = new TFBuffer;
+      buf->len_ = input[i].size;
+      buf->data_ = input[i].data;
+
+      TF_DataType dataType = getTensorTypeToTF_Capi (input[i].type);
+
+      if (dataType == TF_RESOURCE){
+        g_critical ("This data type is not valid: %d", input[i].type);
+        return -1;
+      }
+
+      in = TensorCApi::MakeTensor (
+        dataType,
+        input_tensor_info[i].shape,
+        buf
+      );
+      if (!in.IsAligned ()) {
+        g_critical ("the input tensor %s is not aligned", inputTensorMeta.info[i].name);
+        return -2;
```

Comment:
Does it OK without **delete buf**?

Reply:
@wooksong 
I checked the memory leak through valgrind.
Thanks for your detection.
It will be updated :)